autoemulate/core/compare.py

```diff
@@ -620,13 +629,18 @@ def plot(  # noqa: PLR0912, PLR0913, PLR0915
         test_x, test_y = self._convert_to_tensors(self.test)
 
         # Re-run prediction with just this model to get the predictions
-        y_pred = model.predict(test_x)
-        y_variance = None
-        if isinstance(y_pred, DistributionLike):
-            y_variance = y_pred.variance
-            y_pred = y_pred.mean
+        y_pred, y_variance = (
+            model.predict_mean_and_variance(test_x)
+            if isinstance(model, ProbabilisticEmulator)
```

Comment:
Could this be simplified to a single if condition that checks if `model.supports_uq`? This should evaluate to True for both `ProbabilisticEmulator` and a `TransformedEmulator` that has a `ProbabilisticEmulator` model inside it.

Reply:
I prefer this option though there is an issue with the typing when implemented like this (since model can be an `Emulator` and this doesn't have the `predict_mean_and_variance` method. An option could be to add the method to the `Emulator` base class and then `return mean, None` for the `DeterminsticEmulator` subclass?

Reply:
Implemented the change to the base class in [4402c3d](https://github.com/alan-turing-institute/autoemulate/pull/755/commits/4402c3d9bf76f52b4fbbcaf3a5ec7a31d5f3faa2) and revised the condition in c71afa3 - I think this improves things for downstream usage, what do you think?

Reply:
I was wondering about returning (mean, None) too. I think I'd be in favour of that. If the user tries to do something with a None variance it will error anyways downstream so it won't fail silently, but it allows us to write more general code.